contracts/utils/Whitelist.sol

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.6.0 <0.8.0;
+
+import "@openzeppelin/contracts/access/AccessControl.sol";
+
+/**
+    @title Whitelist
+    @dev Allows developers to implement whitelisting functionality
+    @dev Inherits methods and variables from AccessControl
+    @dev WHITELISTER_ROLE is used to grant permissions for whitelist control
+
+    Use `onlyWhitelisted` modifier for a method in order to restrict access
+    only for whitelisted addresses:
+
+    ```
+        contract MyContract is Whitelist {
+            function deposit() external onlyWhitelisted {
+                // allow execution only to addresses that are in the whitelist
+            }
+        }
+    ```
+
+    Addresses can be added and removed dynamically via the `addToWhitelist`
+    and `removeFromWhitelist` functions. You can check if an address is
+    whitelisted or not by calling `isWhitelisted` function.
+
+    The entire whitelist functionality can be enabled or disabled via the
+    `enableWhitelist` and `disableWhitelist` functions. You can check if the
+    feature is enabled by calling `isWhitelistEnabled` function.
+
+    The following events are emitted: Enabled, Disabled, AddressAdded, AddressRemoved
+*/
+
+contract Whitelist is AccessControl {
+    // See @openzeppelin/contracts/utils/EnumerableSet.sol for more info
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    /// @dev Emitted when whitelist feature is enabled
+    event Enabled(address senderAddress);
```

Comment:
@naure, what do you mean? 

Here's how events are displayed:

![image](https://user-images.githubusercontent.com/29141708/117967134-61c2f900-b32d-11eb-9f41-c36ba8ec1add.png)
![image](https://user-images.githubusercontent.com/29141708/117968111-81a6ec80-b32e-11eb-9a8a-704fdabd34ef.png)
![image](https://user-images.githubusercontent.com/29141708/117967991-5e7c3d00-b32e-11eb-8fed-36e9bff60e9b.png)

What would you suggest we call it?
`Whitelist.EnableFeature`, `Whitelist.WhitelistEnable` ?

Reply:
I can't think of a more specific appropriate name.

Reply:
The name of the contract ("Whitelist") is not included in the event (it’s just "Enabled"). In these screenshots, the UI only guesses which one is it ("possible interpretations"). Of course it is not so bad because you search events by contract address+event, so it is not ambiguous.

Reply:
It’s ok to keep it that way.